REST/LinkedInClient.php

```diff
@@ -78,29 +79,31 @@ public function connect($appKey, $appSecret, $accessToken, $tokenSecret){
         }
     }
 
-    private function getConnectionByApplication()
+    /**
+     * Return a connection based on the suplied Token
+     *
+     * @param Token $token
+     *
+     * @return Client
+     */
+    private function getConnectionByToken(Token $token)
     {
         $application = $this->oauthApp
             ->getApplication(self::RESOURCE_OWNER);
 
-        $token = $this->oauthToken
-            ->getTokenByApplication($application);
-
-        if (!$token) {
-            return null;
-        }
-
         return $this->connect($application->getKey(), $application->getSecret(), $token->getAccessToken(), $token->getTokenSecret());
     }
 
     /**
      * Return the available companies
      *
+     * @param Token $token
+     *
      * @return array
      */
-    public function getCompanies()
+    public function getCompanies(Token $token)
```

Comment:
@ghostika Please stick to the behavior of this class, i.e. you'd first call getConnectionByToken($token), which then returns the connection object and then you execute the getCompanies() method on the connection object. This allows for a clear separation between the method establishing the client connection and the request methods for REST API endpoints.